### PR TITLE
Home run bat

### PR DESCRIPTION
**Pull request by musicmanvr**
## About the PR

Adds the Syndicate Home Run Bat, a highly unusual weapon designed only the craziest of operatives.

On hit, it launches a target like a gorilla gaunlet, but at triple the speed, doing 30 brute while also doing heavy structural damage.

## Why / Balance

Syndicate melee weaponry in the high TC territory has always been a problem, so I figured I'd throw my hat into the ring and give my own take on a high TC melee weapon a shot.

## Technical details

It's a new yaml file with some new sprites, that's all.

## Media

- [ X ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showca

https://github.com/DeltaV-Station/Delta-v/assets/54725557/fcfb117f-fe57-487d-ab3c-7573ad8b18b1

## Breaking changes

Simple Yaml, happy times.

**Changelog**

:cl:
- add: Added the Home Run Bat for Syndicate operatives. Send your foes flying!

### CODE REVIEW

**Comment by deltanedas** [comment]
goofy as fuck i love it

File: Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml]: will this unanchor nukes or 500 glimmer probers :warning: 

also yaml typos:
Speed -> speed
Lifetime -> lifetime,
Enabled just remove it since its true by default

Comment (musicmanvr) [reply Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml]: It uses the same properties as the regular gorilla glove, but lemme test

Comment (musicmanvr) [reply Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml]: > will this unanchor nukes or 500 glimmer probers :warning: 
> 
> 
> 
> also yaml typos:
> 
> Speed -> speed
> 
> Lifetime -> lifetime,
> 
> Enabled just remove it since its true by default

Congratulations, you just discovered a bug with the gorilla gauntlet. The gauntlet can unanchor glimmer probers at 500, which means the bat can too.

Comment (deltanedas) [reply Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml]: prober should prevent unanchor via canceling an event or something on top of the shock thing when using wrench

Comment (musicmanvr) [reply Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml]: > prober should prevent unanchor via canceling an event or something on top of the shock thing when using wrench

I mean I can try to code it to where if glimmer is over 500 it physically cannot be unanchored

Comment (musicmanvr) [reply Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml]: but at the end of the day that'd also make any unanchor attempts whatsoever not do the special shock thing

Comment (musicmanvr) [reply Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml]: > prober should prevent unanchor via canceling an event or something on top of the shock thing when using wrench

i think i have a fix, if i add a line that checks if it's unachored somehow when glimmer is over 500, it'll explode

Comment (deltanedas) [reply Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml]: gorilla gauntlet / home run bat + prober wombo combo in sec maints :trollface:

Comment (DebugOk) [reply Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml]: image

I'll pr a fix for probers soon. Apparently a lot of other things could fling them too, like gravity anomalies

Comment (deltanedas) [reply Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml]: :skull:

Comment (DebugOk) [reply Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml]: For whatever reason that event isn't behaving like how I expected, so it might require a bit more work

Comment (deltanedas) [reply Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml]: you need to set Cancelled = true;

Handled just means it wont use comp.Enabled

Comment (DebugOk) [reply Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml]: Yeah, I know. But this event isn't getting triggered at all.

Comment (DebugOk) [reply Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml]: Ah, the event is getting raised on the owner of the MeleeThrowOnHit component, not the thing being hit. Guess I'll just add a new event

Comment (deltanedas) [reply Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml]: or just add another field (if the weapon isnt stored already it might be) for flexibility and raise it on the thing being hit as well, less logic to maintain

Comment (DebugOk) [reply Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml]: Oh yeah, guess that works too

Comment (DebugOk) [reply Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml]: #824

**Review by DebugOk — COMMENTED** [review]
I'm still unsure about this, it might be abused heavily. But I'll leave it up to the judgement of the other maintainers

**Comment by VMSolidus** [comment]
I think I would prefer it if this bat wasn't 2.5x as powerful as a wielded baseball bat, even when unwielded. Maybe a little bit stronger would be fine, but not to the point that you could very easily crack open the vault with it. I would like this a bit more of it was something you'd buy for its utility, rather than being arguably one of the best melee weapons in the game. Having knockback on a swing is a very cool utility for sure. 

If I ever revisited this later, I actually would want to reflavor it as a "Gorlex Breacher Fist", a power fist sort of weapon that shares its color scheme and design queues with a bloodred hardsuit.

File: Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml
Comment (VMSolidus) [inline Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml]: We need to take a look at toning these numbers back. I don't think that this should be a solid argument for the best melee weapon in the syndicate. It would be better off being balanced around paying for the knockback utility than for raw damage output.

**Review by VMSolidus — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml
Comment (musicmanvr) [reply Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml]: i wrote those damage values when sleep deprived my b

Comment (musicmanvr) [reply Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml]: it's been toned down significantly

File: Resources/Prototypes/DeltaV/Catalog/uplink_catalog.yml
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/Catalog/uplink_catalog.yml:23]: sus

File: Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Melee/bat_home_run.yml:5]: why cant this parent bat and remove like 75% of components

**Review by deltanedas — CHANGES_REQUESTED** [review]
have to fix test too, merging master either now or after upstream pull should do it for the uplink categories

**Comment by deltanedas** [comment]
@musicmanvr are you still working on this

**Pull request closed without merging**
